src/components/FlexibleForm/ControlDecorators/partials/OptionsList.js

```diff
@@ -51,12 +55,12 @@ export const OptionsList = ({
                 role="menuitem"
                 align="start"
                 onClick={() => {
-                  onSelect(option?.[optionValue] || option?.[optionLabel]);
+                  onSelect(getAcceptedValuesOption(option, optionToSelect, 'code'));
```

Comment:
Same here. Option list has not to know about 'code', it should be flexible. Maybe it makes sense to initialize options values outside and pass it to this component?